src/export/StructureDefinitionExporter.ts

```diff
@@ -105,47 +91,54 @@ export class StructureDefinitionExporter implements Fishable {
       throw new ParentNotProvidedError(fshDefinition.name, fshDefinition.sourceInfo);
     }
 
-    let defnType: Type;
-    if (fshDefinition instanceof Extension) {
-      defnType = Type.Extension;
-    } else if (fshDefinition instanceof Logical) {
-      defnType = Type.Logical;
-    } else {
-      defnType = Type.Resource;
+    // We want to ensure that the FSH definition does not set either the name or the id
+    // to be the same value as the parent. Additionally, we want to provide a helpful
+    // error message when this condition exists by suggesting the use of the parent's URL
+    // if we can by using fishForMetadata() for the parent. The caveat is that we cannot
+    // fish for a type that is the same as the FSH definition (e.g., cannot fish for
+    // Type.Profile when FSH definition is an instanceof Profile) because the result of
+    // the fishing would only find itself.
+
+    let possibleParentMeta: Metadata;
+    if (fshDefinition instanceof Logical) {
+      // Logical models can have resources, complex-types, and other logical models as a parent
+      possibleParentMeta = this.fishForMetadata(fshDefinition.parent, Type.Resource, Type.Type);
+    } else if (fshDefinition instanceof Profile) {
+      // Profiles can have resources and other profiles as a parent
+      possibleParentMeta = this.fishForMetadata(fshDefinition.parent, Type.Resource);
```

Comment:
This cleaned things up nicely.  Thank you!

I do have one small concern in that we are now doing this fishing operation for every logical model and every profile, regardless of whether or not their name/id matches the parent.  In the very large majority of cases, there isn't a match, so this is a lot of unnecessary fishing.

We should probably wrap this whole section (lines 102 - 128) in an `if` block that first checks if the parent matches the name or id.

Reply:
Yep... You did mention that and I forgot about it. The above reminder about profiles having a type parent allowed resulted in further cleanup...